bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/AbstractDiscoveryService.java

```diff
@@ -454,4 +439,49 @@ private boolean getAutoDiscoveryEnabled(Object autoDiscoveryEnabled) {
     private String inferKey(DiscoveryResult discoveryResult, String lastSegment) {
         return "discovery." + discoveryResult.getThingUID().getAsString().replaceAll(":", ".") + "." + lastSegment;
     }
+
+    protected DiscoveryResult getLocalizedDiscoveryResult(final DiscoveryResult discoveryResult,
+            @Nullable Bundle bundle) {
+        if (i18nProvider != null && localeProvider != null && bundle != null) {
+            String defaultLabel = discoveryResult.getLabel();
+
+            String key = I18nUtil.stripConstantOr(defaultLabel, () -> inferKey(discoveryResult, "label"));
+
+            ParsedKey parsedKey = new ParsedKey(key);
+
+            String label = i18nProvider.getText(bundle, parsedKey.key, defaultLabel, localeProvider.getLocale(),
+                    parsedKey.args);
+
+            return DiscoveryResultBuilder.create(discoveryResult.getThingUID())
+                    .withThingType(discoveryResult.getThingTypeUID()).withBridge(discoveryResult.getBridgeUID())
+                    .withProperties(discoveryResult.getProperties())
+                    .withRepresentationProperty(discoveryResult.getRepresentationProperty()).withLabel(label)
+                    .withTTL(discoveryResult.getTimeToLive()).build();
+        } else {
+            return discoveryResult;
+        }
+    }
+
+    /**
+     * Utility class to parse the key with parameters into the key and optional arguments.
+     */
+    private final class ParsedKey {
+
+        private static final int LIMIT = 2;
+
+        private final String key;
+        private final Object @Nullable [] args;
+
+        private ParsedKey(String label) {
+            String[] parts = label.split("\\s+", LIMIT);
+            this.key = parts[0];
+
+            if (parts.length == 1) {
+                this.args = null;
+            } else {
+                this.args = Arrays.stream(parts[1].replaceAll("\\[|\\]|\"", "").split(","))
+                        .filter(s -> s != null && !s.trim().isEmpty()).map(s -> s.trim()).toArray(Object[]::new);
```

Comment:
```suggestion
                        .filter(s -> s != null && !s.isBlank()).map(s -> s.trim()).toArray(Object[]::new);
```

Reply:
Note that I take this method from another existing class.

Reply:
Updated

Reply:
Thanks.

> Note that I take this method from another existing class.

Might be an older Java leftover. The `isBlank()` method wasn't available before Java 11.